.github/workflows/test.yml

```diff
@@ -0,0 +1,61 @@
+name: Test
+
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-20.04]
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+
+      # Setup requirements
+      - name: Cache pip packages
+        uses: actions/cache@v2
+        id: cache-venv
+        with:
+          path: ./venv
+          key: cache-venv-${{ matrix.runs-on }}-01
+      - name: Run pip install
+        run: |
+          python -m venv ./venv && . ./venv/bin/activate
+          pip install -r requirements.txt
+        if: steps.cache-venv.outputs.cache-hit != 'true'
+
+      # Run tests
+      - name: Run pytest
+        run: |
+          . ./venv/bin/activate
+          pytest
+
+      # Test upload
+      - run: cat > dummy-scm-1.rockspec <<<
+          "package = 'dummy' version = 'scm-1'"
+      - name: Run rocks server
+        run: |
+          . ./venv/bin/activate
+          python - << '__SCRIPT'
+          import os, sys, app, tests.conftest
+          setattr(app.boto3, 'client', tests.conftest.S3Mock)
+          setattr(app, 'USER', 'test')
+          setattr(app, 'PASSWORD', 'test')
+
+          if os.fork() > 0: # daemonize
```

Comment:
But after a test you should terminate your application.

I suggest something like 'pid = `python ... &` .. kill -9 $pid"

Reply:
It's a Github runner, I guess it's cleaned automatically. Moreover, nothing like `kill -9` will be executed if the job fails. The only way to setup a post job is to add another local composite action, but it seems to be an overkill.

Reply:
It's an implementation detail. One day it could be a self-hosted runner where you should terminate all your tasks manually.